### PR TITLE
Added log message around the distribution failure

### PR DESCRIPTION
**Pull request by saidu941**
Added more logging around the failure while comparing the two slices

### CODE REVIEW

File: pkg/string/slice_test.go [hunk 1]
```diff
@@ -207,8 +208,8 @@ func TestCompareTwoSlices(t *testing.T) {
 	slice3 := []string{"s1", "s2", "s4"}
 	slice4 := []string{"s1", "s2"}
 
-	assert.True(CompareTwoSlices(slice1, slice2))
-	assert.False(CompareTwoSlices(slice1, slice3))
-	assert.False(CompareTwoSlices(slice2, slice3))
-	assert.False(CompareTwoSlices(slice1, slice4))
+	assert.True(CompareTwoSlices(vzlog.DefaultLogger(), slice1, slice2))
```
Comment (anders-swanson) [inline pkg/string/slice_test.go:211]: you can create logger once, if you like. `logger := vzlog.DefaultLogger()` then use `logger` in each call.
Please also note that this pattern you are using can be a parameterized test.

File: pkg/string/slice.go [hunk 2]
```diff
@@ -67,15 +69,19 @@ func SliceAddString(slice []string, s string) ([]string, bool) {
 }
 
 // compareSlices compares 2 string slices after sorting
-func CompareTwoSlices(slice1 []string, slice2 []string) bool {
+func CompareTwoSlices(log vzlog.VerrazzanoLogger, slice1 []string, slice2 []string) bool {
 	sort.Strings(slice1)
```
Comment (anders-swanson) [inline pkg/string/slice.go:73]: You should be copying the slices before sorting them. This is going to mutate the input parameters, perhaps unexpectedly.

File: pkg/string/slice.go [hunk 3]
```diff
@@ -67,15 +69,19 @@ func SliceAddString(slice []string, s string) ([]string, bool) {
 }
 
 // compareSlices compares 2 string slices after sorting
-func CompareTwoSlices(slice1 []string, slice2 []string) bool {
+func CompareTwoSlices(log vzlog.VerrazzanoLogger, slice1 []string, slice2 []string) bool {
```
Comment (anders-swanson) [inline pkg/string/slice.go:72]: Can you call this function AreSlicesEqualWithoutOrder or something similar. I don't know what `compare` means just looking at the function signature.

File: pkg/string/slice_test.go (hunk 1 above)
Comment (saidu941) [reply pkg/string/slice_test.go]: Addressed and updated PR to use log from test framework

File: pkg/string/slice.go (hunk 2 above)
Comment (saidu941) [reply pkg/string/slice.go]: Addressed and updated PR

File: pkg/string/slice.go [hunk 4]
```diff
@@ -67,15 +69,19 @@ func SliceAddString(slice []string, s string) ([]string, bool) {
...
 	sort.Strings(slice1)
 	sort.Strings(slice2)
 
 	if len(slice1) != len(slice2) {
+		log.Errorf("%s doesn't match with %s", slice1, slice2)
```
[the remark this answers is not in the document]
Comment (saidu941) [reply pkg/string/slice.go]: Addressed and updated PR to use log from test framework

File: pkg/string/slice.go (hunk 3 above)
Comment (saidu941) [reply pkg/string/slice.go]: Addressed and updated PR

File: pkg/string/slice.go [hunk 5]
```diff
@@ -67,15 +67,19 @@ func SliceAddString(slice []string, s string) ([]string, bool) {
...
+func AreSlicesEqualWithoutOrder(slice1 []string, slice2 []string) bool {
+	s1 := make([]string, len(slice1))
+	s2 := make([]string, len(slice2))
+	copy(s1, slice1)
+	copy(s2, slice2)
+	sort.Strings(s1)
+	sort.Strings(s2)
 
-	if len(slice1) != len(slice2) {
+	if len(s1) != len(s2) {
```
Comment (anders-swanson) [inline pkg/string/slice.go:78]: you could move this check before the copying/sorting as an optimization.

**Review by anders-swanson — DISMISSED** [review]

File: pkg/string/slice.go (hunk 5 above)
Comment (saidu941) [reply pkg/string/slice.go:76]: Addressed and updated PR

**Review by anders-swanson — APPROVED** [review]

**Pull request merged**